docs/explanations/why-multiprocessing.rst

```diff
@@ -0,0 +1,39 @@
+Why use Multiprocessing?
```

Comment:
I think we should link to https://github.com/DiamondLightSource/python-hdf5-reader-service/issues/8 from this file, one sentence is fine